python/cudf/cudf/core/reshape.py

```diff
@@ -681,3 +682,215 @@ def merge_sorted(
     )
     result._copy_categories(objs[0])
     return result
+
+
+def _pivot(df, index, columns):
+    """
+    Reorganize the values of the DataFrame according to the given
+    index and columns.
+
+    Parameters
+    ----------
+    df : DataFrame
+    index : cudf.core.index.Index
+        Index labels of the result
+    columns : cudf.core.index.Index
+        Column labels of the result
+    """
+    columns_labels, columns_idx = columns._encode()
+    index_labels, index_idx = index._encode()
+    column_labels = columns_labels.to_pandas().to_flat_index()
+
+    # the result of pivot always has a multicolumn
+    result = cudf.core.column_accessor.ColumnAccessor(
+        multiindex=True, level_names=(None,) + columns._data.names
+    )
+
+    def as_tuple(x):
+        return x if isinstance(x, tuple) else (x,)
```

Comment:
Can we just move this out of the `_pivot` function into say `_as_tuple`. No need to redefine this ever invocation of `_pivot`

Reply:
Keeping it within the function maintains the intent that it's meant to be used only from within this function. If we define it elsewhere then it will end up being used in other functions.

The cost of having a nested function like this is non-zero but it's not the same as defining a function "from scratch" -- it's [really small](https://stackoverflow.com/questions/7839632/is-there-an-overhead-when-nesting-functions-in-python/50164012#50164012).